datasets/parsinlu_reading_comprehension/parsinlu_reading_comprehension.py

```diff
@@ -0,0 +1,143 @@
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = (
+        "parsinlu-repo"  # It's not mandatory to have a default configuration. Just use one if it make sense.
+    )
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "question": datasets.Value("string"),
+                "url": datasets.Value("string"),
+                "passage": datasets.Value("string"),
+                "answers": datasets.features.Sequence(
+                    {
+                        "answer_start": datasets.Value("int32"),
+                        "answer_text": datasets.Value("string"),
+                    }
+                ),
+            }
+        )
+
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        data_dir = dl_manager.download_and_extract(_URLs)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": data_dir["train"],
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"filepath": data_dir["test"], "split": "test"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": data_dir["dev"],
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split):
+        logger.info("generating examples from = %s", filepath)
+
+        with open(filepath, encoding="utf-8") as f:
+            for id_, row in enumerate(f):
+                data = json.loads(row)
+                answer = data["answers"]
+                if type(answer[0]) == str:
+                    answer = [{"answer_start": -1, "answer_text": x} for x in answer]
```

Comment:
Do you think it would be possible to find the missing `answer_start` from the passage ?
Maybe by doing something like this ?
```python
answer = [{"answer_start": data["passage"].index(x), "answer_text": x} for x in answer]
```

Reply:
done.